### PR TITLE
experimental flag to run Jenkins without YUI

### PR DESCRIPTION
**Pull request by mawinter69**
The YUI library is old and no longer maintained.

Add a user experimental flag to disable YUI. The flag is disabled by default. When enabling the flag, all the YUI related js libraries and css classes are not loaded.

Following PR are required to get Jenkins to not show any errors
#9488
#9453
#7569

Some plugins that use YUI (not complete):
credentials
ldap
global-build-stats
build-monitor
categorized-view
matrix-project

Plugins that make use of makeButton (not complete) 
credentials (fixed with https://github.com/jenkinsci/credentials-plugin/pull/533)
openid

acceptance-test-harness

### Proposed changelog entries

Add a user experimental flag to run Jenkins without the YUI JavaScript library. Plugin authors should enable this flag and fix any issues that result from the removal of the YUI library.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [ ] The Jira issue, if it exists, is well-described.
- [ ] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [ ] There is automated testing or an explanation as to why this change has no tests.
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

File: core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java
Comment (janfaracik) [inline core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java:9]:
```suggestion
        super("remove-yui.flag");
```

**Review by janfaracik — APPROVED** [review]

**Comment by janfaracik** [comment]
Looks good to me! Thanks for compiling that list.

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java:1]: Please add license header

File: core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java:6]:
```suggestion
@Extension
@Restricted(NoExternalUse.class)
```

plus imports

File: core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java
Comment (mawinter69) [reply core/src/main/java/jenkins/model/experimentalflags/RemoveYuiUserExperimentalFlag.java:25]: done

**Comment by daniel-beck** [comment]
> Some plugins that use YUI (not complete):

Add matrix-auth and azure-ad to the list due to their use of `YAHOO.lang`.

**Comment by mawinter69** [comment]
I've collected all usages of YUI here:
https://docs.google.com/spreadsheets/d/1UjvtFmNmEdjMN5DUoFxJfBryA8q-E5_HwOzVKbVG9b0/edit?usp=sharing

**Pull request merged**
